File: inventory/src/main/java/com/gqlfederationexample/inventory/ProductService.java
package com.gqlfederationexample.inventory;

import org.springframework.stereotype.Service;

import java.math.BigInteger;
import java.security.MessageDigest;
import java.security.NoSuchAlgorithmException;

@Service
public class ProductService {
    public Product lookupProduct(String upc) {
        try {
            // Why not?
            int quantity = Math.floorMod(
                    new BigInteger(1,
                            MessageDigest.getInstance("SHA1").digest(upc.getBytes())
                    ).intValue(),
                    10_000);

            return new Product(upc, quantity);
        } catch (NoSuchAlgorithmException e) {
            throw new RuntimeException(e);
        }
    }
}

File: inventory/src/main/java/com/gqlfederationexample/inventory/domain/service/ProductService.kt
package com.gqlfederationexample.inventory.domain.service

import com.gqlfederationexample.inventory.domain.model.Product
import org.springframework.stereotype.Service
import java.math.BigInteger
import java.security.MessageDigest
import java.security.NoSuchAlgorithmException

@Service
class ProductService {
    fun lookupProduct(upc: String): Product {
        return try {
            // Why not?
            val quantity = Math.floorMod(
                BigInteger(
                    1,
                    MessageDigest.getInstance("SHA1").digest(upc.toByteArray())
                ).toInt(),
                10000
            )
            Product(upc, quantity)
        } catch (e: NoSuchAlgorithmException) {
            throw RuntimeException(e)
        }
    }
}